pkg/plugin/plugin.go

```diff
@@ -0,0 +1,113 @@
+package plugin
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+	"syscall"
+)
+
+// Handler is capable of parsing command line arguments
+// and performing executable filename lookups to search
+// for valid plugin files, and execute found plugins.
+type Handler interface {
+	// exists at the given filename, or a boolean false.
+	// Lookup will iterate over a list of given prefixes
+	// in order to recognize valid plugin filenames.
+	// The first filepath to match a prefix is returned.
+	Lookup(filename string) (string, bool)
+	// Execute receives an executable's filepath, a slice
+	// of arguments, and a slice of environment variables
+	// to relay to the executable.
+	Execute(executablePath string, cmdArgs, environment []string) error
+}
+
+// DefaultPluginHandler implements PluginHandler
```

Comment:
```suggestion
// DefaultPluginHandler implements Handler
```
also, given that we're inside pkg/plugin, from outside these are accessed as `plugin.DefaultPluginHandler` and `plugin.NewDefaultPluginHandler()`, you could drop Plugin from the name.  not critical.